homeassistant/components/mychevy.py

```diff
@@ -0,0 +1,343 @@
+    _unit = "miles"
+    _name = "EV Sensor"
+    _attr = None
+
+    def __init__(self, connection):
+        """Initialize sensor with car connection."""
+        self._conn = connection
+        self.hass = connection.hass
+        self.car = connection.car
+        self.entity_id = ENTITY_ID_FORMAT.format(
+            '{}_{}'.format(DOMAIN, self.__class__.__name__))
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {"units": self._unit}
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state."""
+        if self.car is not None:
+            return getattr(self.car, self._attr, None)
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement the state is expressed in."""
+        return self._unit
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+
+class EVCharge(EVSensor):
+    """EVCharge Sensor.
+
+    The charge percentage of the EV battery. It is an integer from 0 -
+    100.
+
+    """
+
+    _name = "EV Charge"
+    _unit = "percent"
+    _attr = "percent"
+
+
+class EVMileage(EVSensor):
+    """EVMileage Sensor.
+
+    The total odometer mileage on the car.
+    """
+
+    _name = "EV Mileage"
+    _unit = "miles"
+    _attr = "mileage"
+
+
+class EVRange(EVSensor):
+    """EV Range.
+
+    The estimated average range of the car. This is going to depend on
+    both charge percentage as well as recent driving conditions (for
+    instance very cold temperatures drive down the range quite a bit).
+
+    """
+
+    _name = "EST Range"
+    _unit = "miles"
+    _attr = "range"
+
+
+class EVPlugged(EVSensor):
+    """EV Plugged in.
+
+    Is the EV Plugged in. Returns True or False. This does not
+    indicate whether or not it's currently charging.
+
+    """
+
+    _name = "EV Plugged In"
+    _unit = None
+    _attr = "plugged_in"
+
+
+class EVCharging(EVSensor):
+    """A string representing the charging state."""
+
+    _name = "EV Charging"
+    _unit = None
+    _attr = "charging"
+
+
+class EVChargeMode(EVSensor):
+    """A string representing the charging mode."""
+
+    _name = "EV Charge Mode"
+    _unit = None
+    _attr = "charge_mode"
+
+
+class MyChevyStatus(Entity):
+    """A string representing the charge mode."""
+
+    _name = "MyChevy Status"
+    _icon = None
+
+    def __init__(self, connection):
+        """Initialize sensor with car connection."""
+        self._state = MYCHEVY_UNKNOWN
+        self._last_update = dt.now()
+        self._conn = connection
+        self.hass = connection.hass
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {"last_update": self._last_update}
```

Comment:
Time stamp is not allowed in the state attributes. Only a time change should not create a state update.

Reply:
ack